File: kraken/src/main/java/glitch/kraken/object/json/list/Games.java
package glitch.kraken.object.json.list;

import com.google.common.collect.ImmutableList;
import com.google.gson.annotations.SerializedName;
import glitch.api.objects.json.interfaces.OrdinalList;
import glitch.kraken.object.json.Game;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
public class Games implements OrdinalList<Game> {
    private final ImmutableList<Game> data;
    @SerializedName("_total")
    @Getter(AccessLevel.NONE)
    private final int total;

    @Override
    public int size() {
        return total;
    }
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/collections/Games.kt
package glitch.kraken.`object`.json.collections

import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.OrdinalList
import glitch.kraken.`object`.json.Game

data class Games(
        override val data: List<Game>,
        @SerializedName("_total")
        override val size: Int = data.size
) : OrdinalList<Game>